pkg-manager/resolve-dependencies/src/resolveDependencies.ts

```diff
@@ -1565,18 +1568,30 @@ async function resolveDependency (
       }
     }
   }
+
+  let normalizedBareSpecifier = pkgResponse.body.normalizedBareSpecifier
+  const resolvedPkg = ctx.resolvedPkgsById[pkgResponse.body.id]
+  const alias = wantedDependency.alias ?? pkgResponse.body.alias ?? pkg.name
+  if (alias && ctx.saveCatalog && wantedDependency.source === 'cli-param' && normalizedBareSpecifier && resolvedPkg != null) {
+    ctx.addNewDefaultCatalog(alias, {
+      specifier: normalizedBareSpecifier,
+      version: resolvedPkg.version,
+    })
+    normalizedBareSpecifier = 'catalog:'
+  }
```

Comment:
This code doesn't makes sense here because it is only needed for direct dependencies. But this function runs for all dependencies. We can take catalog versions in resolveDependencyTree after resolution from the return direct dependencies.

Reply:
https://github.com/pnpm/pnpm/pull/9484/commits/04b6f5522aafa11cd7bee4b1cc7a7f448182fc34